app/models/position.rb

```diff
@@ -180,6 +180,21 @@ def remove
     false
   end
 
+  def placements_to_disqualify_on_closing
+    placements
+      .joins(:position_stage)
+      .where.not(status: "disqualified")
+      .where.not(position_stage: { name: "Hired" })
+  end
+
+  def placements_to_requalify_on_reopening
+    placements
+      .joins(:position_stage)
+      .where(status: "disqualified")
+      .where.not(position_stage: { name: "Hired" })
+      .includes(:candidate)
```

Comment:
nitpick: I'd suggest to move `includes` to the place where the method calls, since we are not 100% sure that need to preload candidates every time on calling method